tools/build.py

```diff
@@ -126,6 +126,8 @@ def devhelp(helpstring):
                          help='specify profile file')
     coregrp.add_argument('--regexp-strict-mode', metavar='X', choices=['ON', 'OFF'], type=str.upper,
                          help=devhelp('enable regexp strict mode (%(choices)s)'))
+    coregrp.add_argument('--regexp-recursion-limit', metavar='N', type=int, default=1000,
+                         help='RegExp recursion depth limit (0 for unlimited, default=1000)')
```

Comment:
I have doubts about this: 1) defaults have been removed from build.py for a while to avoid the need for maintaining defaults at multiple places (build.py and cmakelists) or their potential divergence, 2) this would change the existing default of no recursion limit to a set limit (with both code size and performance implications), and 3) the new default (1000) is completely arbitrary.

I'd vote for no change of current behavior, only opt-in for recursion limit.

(Moreover: help string should use all-lowercase "regexp" as that is used in existing help messages.)

Reply:
I will remove the default and change the argument to optional.